uds/translator/configurable_translator.py

```
@@ -0,0 +1,1002 @@
+
+        :return: Conditional Data Record for DID data.
+        """
+        default_did_data = RawDataRecord(name=name,
+                                         length=8,
+                                         min_occurrences=1,
+                                         max_occurrences=None)
+
+        def _get_did_data(did: int) -> tuple[RawDataRecord]:
+            data_records = self.did_data_mapping.get(did, None)
+            if data_records is None:
+                raise ValueError(f"No data structure defined for DID 0x{did:04X}.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)